models/db.py

```diff
@@ -184,7 +184,37 @@ def sanitize_fields(form):
     for site in current.SITES:
         site_handle = site.lower() + "_handle"
         form.vars[site_handle] = form.vars[site_handle].lower()
-
+#-----------------------------------------------------------------------------
+def inside_institute(form):
+        # Send mail to all users form the same college when someone registers
+    to = []
+    for mail in db(utable.email == form.vars.institute).select():
```

Comment:
Okay this line says you haven't tested the code.
`utable` not defined - Should be atable, and `atable = db.auth_user`
and the query should be 
```
db(atable.institute == form.vars.institute).select(atable.email)
```
The `mail` variable could be renamed as `email_id`
Finally keep this query in a seperate variable(say varname is `rows`)
And then 
```
for email_id in rows:
```
would be a cleaner way.